ybn/operators.py

```diff
@@ -241,9 +243,16 @@ def run(self, context):
             return False
 
         mat = create_collision_material_from_index(context.window_manager.sz_collision_material_index)
-        active_mat_index = aobj.active_material_index
-        aobj.data.materials[active_mat_index] = mat
-
+        
+        if convert_all:
+            for obj in bpy.data.objects:
+                if obj.type == 'MESH':
+                    for i, slot in enumerate(obj.data.materials):
+                        if slot == obj.active_material:
+                            obj.data.materials[i] = mat
```

Comment:
Should this really be changing the materials of *every* object in the .blend? Including those that are not even collisions? Like, if I understand correctly, any drawable mesh in the .blend will also end up with a collision material.

If anything, this operator should be changed so it also applies to `context.selected_objects` not only `context.active_object`.

Reply:
not every object, only the objects that are sharing that same material

Reply:
but yes, for a more controlled approach you could use the selected objects only, even tho you would need to select them all everytime you want to convert materials

Reply:
> not every object, only the objects that are sharing that same material

That's not what the code does. It replaces the active material of every object.

https://github.com/user-attachments/assets/1852679e-97dc-4d32-a5a0-e7c24a5d82c7

Reply:
what? for me was working fine, I used it to convert entire maps and it converted only the shared materials, maybe some commit broke it. I'll look into it

Reply:
https://github.com/user-attachments/assets/ece80c88-1e54-474e-ad0a-1a5a555dbf6a

okay i fixed it